packages/components/src/internal/renderers/TemplateDownloadRenderer.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { ReactNode } from 'react';
+import { TemplateDownloadButton } from '../../public/files/TemplateDownloadButton';
+import { Map } from 'immutable';
+import { downloadSampleTypeTemplate, getSampleTypeTemplateUrl } from '../components/samples/utils';
+import { SCHEMAS } from '../schemas';
+import { SchemaQuery } from '../../public/SchemaQuery';
+import { Assay } from '@labkey/api';
+import { downloadFromUrl } from '../util/utils';
+
+interface Props {
+    row?: Map<any, any>;
+}
+
+export class SampleTypeTemplateDownloadRenderer extends React.PureComponent<Props> {
+
+    onDownload = () => {
+        const { row } = this.props;
+        const schemaQuery = SchemaQuery.create(SCHEMAS.SAMPLE_SETS.SCHEMA,  row.getIn(['Name', 'value']));
+        downloadSampleTypeTemplate(schemaQuery, getSampleTypeTemplateUrl);
+    }
+
+    render(): ReactNode {
+        return (
+            <TemplateDownloadButton onClick={this.onDownload} text={"Download"} className={"button-small-padding"}/>
+        )
+    }
+}
+
+
+export class AssayResultTemplateDownloadRenderer extends React.PureComponent<Props> {
+
+    onDownload = () => {
+        const { row } = this.props;
+        Assay.getByName({
+            name: row.getIn(['Name', 'value']),
```

Comment:
any need for case insensitive check here on 'Name'?

Reply:
I don't think so, but added just in case.